android/src/cg/group4/sensor/AndroidAccelerationStatus.java

```diff
@@ -67,15 +79,20 @@ public final void movementChanged(final Movement movement) {
         if (movement.name().equals("WALKING")) {
             Gdx.app.debug(tag, "You are walking!");
             cAccelerationState = AccelerationState.WALKING;
+            cUpdateMovementSubject.update(AccelerationState.WALKING);
```

Comment:
At the end of the if statements you could just call `cUpdateMovementSubject.update(cAccelerationState);` once